tools/python/model_validation/validation_tool.py

```diff
@@ -0,0 +1,145 @@
+
+
+        params = og.GeneratorParams(model)
+        params.set_search_options(**search_options)
+        params.input_ids = input_tokens
+
+        generator = og.Generator(model, params)
+        if config["verbose"]: print("Generator created")
+
+        if config["verbose"]: print("Running generation loop ...")
+        if config["timings"]:
+            first = True
+            new_tokens = []
+
+        print()
+        # print("Output: ", end='', flush=True)
+
+        generation_successful = True
+
+        try:
+            while not generator.is_done():
+                generator.compute_logits()
+                generator.generate_next_token()
+                if config["timings"]:
+                    if first:
+                        first_token_timestamp = time.time()
+                        first = False
+
+                new_token = generator.get_next_tokens()[0]
+                
+                value_to_save = tokenizer_stream.decode(new_token)
+
+                complete_text += value_to_save
+
+                # print(tokenizer_stream.decode(new_token), end='', flush=True)
+                
+                if config["timings"]: new_tokens.append(new_token)
+        except KeyboardInterrupt:
+            print("  --control+c pressed, aborting generation--")
+            generation_successful = False
+        except Exception as e:
+            print(f"An error occurred: {e}")
+            generation_successful = False
+
+        print()
+        print()
+
+
+        with open('output.txt', 'a') as file:
+            file.write(complete_text)
+
+        # Delete the generator to free the captured graph for the next generator, if graph capture is enabled
+        del generator
+
+        if config["timings"]:
+            prompt_time = first_token_timestamp - started_timestamp
+            run_time = time.time() - first_token_timestamp
+            print(f"Prompt length: {len(input_tokens)}, New tokens: {len(new_tokens)}, Time to first: {(prompt_time):.2f}s, Prompt tokens per second: {len(input_tokens)/prompt_time:.2f} tps, New tokens per second: {len(new_tokens)/run_time:.2f} tps")
+
+    return generation_successful
+
+def folder_exists(folder_path):
+    return os.path.isdir(folder_path)
+
+def create_folder(folder_path):
+    os.mkdir(folder_path)
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(argument_default=argparse.SUPPRESS, description="End-to-end AI Question/Answer example for gen-ai")
+
+    parser.add_argument('-j', '--json', type=str, required=True, help='Path to the JSON file containing the arguments')
+
+    args = parser.parse_args()
+
+    with open(args.json, 'r') as file:
+        config = json.load(file)
+
+    # Check and see if the folder exists, if not create the folder
+    model_output_dir = "../../../models_outputs/"
+    model_cache_dir = "../../../cache_models"
+
+    if not folder_exists(model_output_dir):
+        create_folder(model_output_dir)
+
```

Comment:
Using `os.makedirs(folder_path, exist_ok=True)` may be a better option here. If `folder_path` exists, then `folder_path` isn't created and the existing `folder_path` is used. If `folder_path` does not exist, then `folder_path` is created. Here is the [documentation](https://docs.python.org/3/library/os.html#os.makedirs) for more info.